File: examples/python/quickstart.py
"""Navigate to a page, read its title, and capture a proof screenshot.

    python examples/python/quickstart.py
"""

from betterwright import BetterWright


def main() -> None:
    with BetterWright() as bw:
        bw.run("await page.goto('https://example.com')", note="Opening example.com")

        title = bw.run("return page.title()")
        print("Title:", title.value)

        proof = bw.run("return screenshot({kind: 'proof', name: 'example-home'})")
        proof.raise_for_status()
        print("Proof:", proof.artifacts[0].media_reference)


if __name__ == "__main__":
    main()

File: examples/typescript/quickstart.ts
// Navigate, read the title, capture a proof screenshot.
//
//   node examples/typescript/quickstart.ts

import { BetterWright, BrowserError } from "betterwright";

const bw = new BetterWright();
try {
  await bw.run("await page.goto('https://example.com')", { note: "Opening example.com" });

  const title = await bw.run("return page.title()");
  if (!title.ok) throw new BrowserError(title.error);
  console.log("Title:", title.result);

  const proof = await bw.run("return screenshot({ kind: 'proof', name: 'example-home' })");
  console.log("Proof:", proof.artifacts?.[0]?.media);
} finally {
  await bw.close();
}
